project_timesheet_time_control/wizards/hr_timesheet_switch.py

```diff
@@ -124,7 +124,7 @@ def action_switch(self):
         ).running_timer_id.button_end_work()
         # Start new timer
         _fields = self.env["account.analytic.line"]._fields.keys()
```

Comment:
@pedrobaeza @konykon What about if we change this line by:

`fields = self.env["account.analytic.line"].fields_get().keys()`

This way we only get fields that user has permission to access.

Reply:
But that way we will lose the information on that fields. I still don't get how to reproduce the problem.

Reply:
Problem is: Module helpdesk_mgmt_timesheet adds 2 fields to account.analytic.line only accessible by group helpdesk_mgmt.group_helpdesk_user. https://github.com/OCA/helpdesk/blob/12.0/helpdesk_mgmt_timesheet/models/hr_timesheet.py

So when we activate this helpdesk module, a non helpdesk user will get permission problems when using project_timesheet_time_control. The call to ```self.env["account.analytic.line"]._fields.keys()```  will return fields you don't have permission to read, create or anything. 

A user not having access to helpdesk but to project should be able to get fields on account.analytic.line he has permission to read. Thats why ```fields_get``` is a better option than ```_fields```. ```fields_get``` already gives you fields you have permission to access.